js_modules/dagster-ui/packages/ui-core/src/assets/AssetWipeDialog.tsx

```diff
@@ -1,75 +1,186 @@
+    isOpen: boolean;
+    onClose: () => void;
+    onComplete: (assetKeys: AssetKey[]) => void;
+    requery?: RefetchQueriesFunction;
+  }) => {
+    return (
+      <Dialog
+        isOpen={props.isOpen}
+        title="Wipe materializations"
+        onClose={props.onClose}
+        style={{width: '80vw', maxWidth: '1200px', minWidth: '600px'}}
+      >
+        <AssetWipeDialogInner {...props} />
+      </Dialog>
+    );
+  },
+);
+
+export const AssetWipeDialogInner = ({
   assetKeys,
-  isOpen,
   onClose,
   onComplete,
   requery,
 }: {
   assetKeys: AssetKey[];
-  isOpen: boolean;
   onClose: () => void;
   onComplete: (assetKeys: AssetKey[]) => void;
   requery?: RefetchQueriesFunction;
 }) => {
   const [requestWipe] = useMutation<AssetWipeMutation, AssetWipeMutationVariables>(
     ASSET_WIPE_MUTATION,
     {
-      variables: {assetKeys: assetKeys.map(asAssetKeyInput)},
       refetchQueries: requery,
     },
   );
 
+  const [isWiping, setIsWiping] = useState(false);
+  const [wiped, setWiped] = useState(0);
+
+  const didCancel = useRef(false);
   const wipe = async () => {
     if (!assetKeys.length) {
       return;
     }
-    await requestWipe();
+    setIsWiping(true);
+    for (let i = 0, l = assetKeys.length; i < l; i += CHUNK_SIZE) {
+      if (didCancel.current) {
+        return;
+      }
+      const nextChunk = assetKeys.slice(i, i + CHUNK_SIZE);
+      const result = await requestWipe({variables: {assetKeys: nextChunk.map(asAssetKeyInput)}});
+      const data = result.data?.wipeAssets;
+      switch (data?.__typename) {
+        case 'AssetNotFoundError':
+          showCustomAlert({
+            title: 'Could not wipe asset materializations',
+            body: 'Asset not found.',
+          });
+          break;
+        case 'AssetWipeSuccess':
+          setWiped((wiped) => wiped + nextChunk.length);
+          break;
+        case 'PythonError':
+          showCustomAlert({
+            title: 'Could not wipe asset materializations',
+            body: <PythonErrorInfo error={data} />,
+          });
+          break;
+        case 'UnauthorizedError':
+          showCustomAlert({
+            title: 'Could not wipe asset materializations',
+            body: 'You do not have permission to do this.',
+          });
+          break;
+      }
+    }
     onComplete(assetKeys);
+    setIsWiping(false);
   };
 
-  return (
-    <Dialog isOpen={isOpen} title="Wipe materializations" onClose={onClose} style={{width: 600}}>
-      <DialogBody>
+  useLayoutEffect(() => {
+    return () => {
+      didCancel.current = true;
+    };
+  }, []);
+
+  const parentRef = useRef<HTMLDivElement>(null);
+
+  const rowVirtualizer = useVirtualizer({
+    count: assetKeys.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 24,
+    overscan: 10,
+  });
+
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  const content = useMemo(() => {
```

Comment:
Is the `useMemo` all that valuable here? Its dependencies are going to change pretty often, no?

Reply:
I mostly am using it for the closure, I think its worth keeping